backend/btrixcloud/operator.py

```diff
@@ -624,6 +626,12 @@ async def update_crawl_state(self, redis, crawl, status):
                 + "gracefully stopping crawl"
             )
 
+        if crawl.max_crawl_size and stats["size"] > crawl.max_crawl_size:
```

Comment:
```suggestion
        if crawl.max_crawl_size and status.size and status.size > crawl.max_crawl_size:
```
Just in case -- stats["size"] may not exist and already updating `status.size`

Reply:
Maybe should move this to be below / as part of this conditional?
```
        if stats["size"] is not None:
            status.size = humanize.naturalsize(stats["size"])
```

Reply:
I moved this up a bit, since we want to set stopping earlier. Go ahead and take a look.

Reply:
Hm, this seems to have broken some tests, I'll take a look in a bit

Reply:
Yeah seems better now. Defaulting back to status.filesAddedSize if stats["size"] isn't set

Reply:
Yep, I think that makes sense - the filesAddedSize should always be <= current size, so that makes sense as a fallback.